tests/unit/backend/layers/business/test_business.py

```diff
@@ -1222,6 +1231,7 @@ def test_get_all_collections_published_does_not_retrieve_old_versions(self):
         """
         first_version = self.initialize_published_collection()
         second_version = self.business_logic.create_collection_version(first_version.collection_id)
+        self.business_logic.remove_dataset_version(second_version.version_id, second_version.datasets[0].version_id)
```

Comment:
The reason for this line is not obvious. Could you add a comment or change the code to be more explicit.